programs/ibc-attestor/src/adapter_impls/optimism.rs

```diff
@@ -1,72 +1,150 @@
 
-        Ok(SyncState(sync_state))
+        sync_state
+            .get("unsafe_l2")
+            .ok_or(AdapterError::UnfinalizedBlockError("no unsafe l2".into()))
+            .map(|block| block.height)
     }
-}
 
-impl Adapter for OpConsensusClient {
-    async fn get_latest_finalized_block(&self) -> Result<impl Signable, AdapterError> {
-        let state = self.get_sync_state().await?;
-
-        state
-            .0
-            .get("finalized_l2")
-            .map(|sync_header| {
-                Header::new(sync_header.height, sync_header.hash, sync_header.timestamp)
+    async fn get_timestamp_for_block_at_height(&self, height: u64) -> Result<u64, AdapterError> {
+        self.client
+            .get_block_by_number(BlockNumberOrTag::Number(height))
+            .await
+            .map_err(|e| AdapterError::FinalizedBlockError(e.to_string()))?
+            .ok_or_else(|| {
+                AdapterError::FinalizedBlockError(format!(
+                    "no OP block of kind {} found",
+                    BlockNumberOrTag::Latest
+                ))
             })
-            .ok_or(AdapterError::FinalizedBlockError(
-                "response received but no finalized L2 found in response".into(),
-            ))
+            .map(|header| header.header.timestamp())
     }
-    async fn get_latest_unfinalized_block(&self) -> Result<impl Signable, AdapterError> {
-        let state = self.get_sync_state().await?;
 
-        state
-            .0
-            .get("unsafe_l2")
-            .map(|sync_header| {
-                Header::new(sync_header.height, sync_header.hash, sync_header.timestamp)
+    async fn get_historical_packet_commitment(
+        &self,
+        packet: &Packet,
+        block_number: u64,
+    ) -> Result<[u8; 32], AdapterError> {
+        let fixed: FixedBytes<32> = packet.commitment_path().as_slice().try_into().unwrap();
+
+        self.router
+            .getCommitment(fixed)
+            .block(BlockId::Number(BlockNumberOrTag::Number(block_number)))
+            .call()
+            .await
+            .map_err(|e| {
+                AdapterError::FinalizedBlockError(format!("Failed to get commitment: {}", e))
             })
-            .ok_or(AdapterError::UnfinalizedBlockError(
-                "response received but no unfinalized L2 found in response".into(),
-            ))
+            .map(|commitment| commitment.into())
     }
-    fn block_time(&self) -> Duration {
-        Duration::from_secs(15)
+}
+
+impl Adapter for OpConsensusClient {
+    async fn get_unsigned_state_attestation_at_height(
+        &self,
+        height: u64,
+    ) -> Result<UnsignedStateAttestation, AdapterError> {
+        let ts = self.get_timestamp_for_block_at_height(height).await?;
+        Ok(UnsignedStateAttestation {
+            height,
+            timestamp: ts,
+        })
+    }
+    async fn get_unsigned_packet_attestation_at_height(
```

Comment:
Not sure if we want to do this, but looking at this, it seems like maybe we could have some common libs for evm-based chains?
Not strictly necessary to do right now, but if we can make the adapter impl even thinner, it would be good for our ability to quickly add new ecosystems!
Could be a separate spike to look into that?